src/pointer-analysis/value_set.cpp

```diff
@@ -194,8 +194,22 @@ void value_sett::output(
         result=from_expr(ns, identifier, o);
       else
       {
-        result="<"+from_expr(ns, identifier, o)+", ";
-
+        if(o.id()==ID_dynamic_object)
+        {
+          dynamic_object_exprt::recencyt recency;
+          std::string recency_str="";
+
+          recency=to_dynamic_object_expr(o).get_recency();
+          if(recency==dynamic_object_exprt::recencyt::MOST_RECENT_ALLOCATION)
+            recency_str="most-recent-allocation";
+          else if(recency==dynamic_object_exprt::recencyt::ANY_ALLOCATION)
+            recency_str="any-allocation";
+          result="<"+from_expr(ns, identifier, o)+", "+recency_str+", ";
+        }
+        else
+        {
+          result="<"+from_expr(ns, identifier, o)+", ";
```

Comment:
If you set this before the if statement then you could append the recency string to the end of it inside the conditional to reduce code repetition.